memo/managers.py

```diff
@@ -73,3 +74,63 @@ def get_card_amount_in_deck(self, deck):
 class DeckManager(models.Manager):
     def get_decks_by_owner(self, user):
         return self.filter(owner=user)
+
+
+class CardAnswerHistoryManager(models.Manager):
+    def get_consecutive_days(self, user):
+        from .models import Card
+
+        max_daily_date_times = (
+            self.filter(owner=user)
+            .extra(select={"day": "date( created_at )"})
+            .values("day")
+            .annotate(latest_datetime=Max("created_at"))
+        )
+
+        objects = (
+            self.filter(
+                created_at__in=[
+                    entry["latest_datetime"] for entry in max_daily_date_times
+                ]
+            )
+            .values("card", "created_at")
+            .order_by("-created_at")
+        )
+
+        result = []
+
+        if objects:
+            answered_today = False
+            today = datetime.today()
+            count = 0
+            one_day_less = 1
+
+            for index in range(0, len(objects)):
+                one_day_before = (today - timedelta(days=one_day_less)).strftime(
+                    "%Y-%m-%d"
+                )
+                created_at = (objects[index].get("created_at")).strftime("%Y-%m-%d")
+
+                if index == 0 and today.strftime("%Y-%m-%d") == created_at:
+                    answered_today = True
+                    count += 1
+                    continue
+
+                if created_at != one_day_before:
+                    break
+
+                count += 1
+                one_day_less += 1
+
+            card = get_object_or_404(Card, pk=objects[0].get("card"))
```

Comment:
```suggestion
            card = get_object_or_404(Card, pk=object[0].card_id)
```

Reply:
This line is not inside of the loop, so the variable is`objects`and this type is a `<class 'django.db.models.query.QuerySet'>` but if I access the first index, it'll give me a `dict` so I can't access the card_id using `.card_id,` still need to use `get("card").`

```
            print(type(objects[0]))
            print(dir(objects[0]))

<class 'dict'>
['__class__', '__class_getitem__', '__contains__', '__delattr__', '__delitem__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__ior__', '__iter__', '__le__', '__len__', '__lt__', '__ne__', '__new__', '__or__', '__reduce__', '__reduce_ex__', '__repr__', '__reversed__', '__ror__', '__setattr__', '__setitem__', '__sizeof__', '__str__', '__subclasshook__', 'clear', 'copy', 'fromkeys', 'get', 'items', 'keys', 'pop', 'popitem', 'setdefault', 'update', 'values']
```

Reply:
Yeah you're right, I must have misspelled when I wrote. 
It becomes a class object (instead of a dict) if you remove the `.values` method from your queryset, but if it messes with the query result, it's ok to leave it. 

(I repeated the answer just to make it clear =] )

Reply:
Thank you, @r-lelis ! I've sent a new commit with the lines you've suggested. Could you check the it, please?